x-pack/plugin/security/qa/basic-enable-security/src/javaRestTest/java/org/elasticsearch/xpack/security/EnableSecurityOnBasicLicenseIT.java

```diff
@@ -58,10 +65,24 @@ protected Settings restClientSettings() {
 
     @Override
     protected boolean preserveClusterUponCompletion() {
-        // If this is the first run (security not yet enabled), then don't clean up afterwards because we want to test restart with data
+        // If this is one of the first two runs (security not yet enabled), then don't clean up afterwards because we want to test restart
+        // with data
         return securityEnabled == false;
     }
 
+    @Override
+    protected RestClient buildClient(Settings settings, HttpHost[] hosts) throws IOException {
+        RestClientBuilder builder = RestClient.builder(hosts);
+        configureClient(builder, settings);
+        if (settings.hasValue("xpack.security.enabled")) {
+            builder.setStrictDeprecationMode(true);
+        } else {
+            builder.setStrictDeprecationMode(false);
+        }
```

Comment:
I've also tested `./gradlew :x-pack:plugin:security:qa:basic-enable-security:check` without:
```
-        if (settings.hasValue("xpack.security.enabled")) {
-            builder.setStrictDeprecationMode(true);
-        } else {
-            builder.setStrictDeprecationMode(false);
-        }
```
and it passes.

Reply:
This is weird. It fails for me. Could you make sure you run against this branch?

Reply:
It passed for me as well. I checked out the PR locally.

Reply:
Do you, guys, run it for all 3 configurations? There should be - security expl. on, expl. off, and impl. off.

Reply:
I just ran the same command as Albert. And here is the buildscan link https://gradle-enterprise.elastic.co/s/e3nie265x74sk

Reply:
@BigPandaToo please note that the diff in my first comment is different from the current version (https://github.com/elastic/elasticsearch/pull/70114/commits/210b37ed263ab045dcf47f73dcf931469910c9cf changed it).

My original point was that `xpack.security.enabled` is not a client setting.

In any case, it works without:
```
-       if (System.getProperty("tests.has_security") != null) {
-           builder.setStrictDeprecationMode(true);
-       } else {
-           builder.setStrictDeprecationMode(false);
-       }
```
too. The reason is that `RestClientBuilder#strictDeprecationMode` is `false` by default.

Reply:
By default `ESRestTestCase` `buildClient` set it to `true`:

`1064>        builder.setStrictDeprecationMode(true);`

unless you override it

Reply:
> I just ran the same command as Albert. And here is the buildscan link https://gradle-enterprise.elastic.co/s/e3nie265x74sk

This really puzzels me. I don't see how it can pass. I think, I have to ask: did you run against this branch (including the actuall change which adds the warning headers to responses)?

Reply:
@albertzaharovits solved the mystery :) You run the test still overriding `buildClient` just leaving deprecation mode to default (which is `false`). In this case all 3 instances are running with the strict deprecation mode `off`, while in my case it is `off` only if security is implicitly disabled, which allows me to verify that we don't have these warning headers when security is explicitly enabled or disabled.